### ISSUE TITLE
Support for IcedCoffeeScript

### ISSUE THREAD

**Issue by niieani** [decaffeinate/decaffeinate#786]
I have a feature request.

<FILL THIS IN WITH DETAILS OF YOUR FEATURE REQUEST>

[IcedCoffeeScript](https://maxtaco.github.io/coffee-script/) is a superset of CoffeeScript and its main addition are two new keywords: `await` and `defer`. It would be great if we could transpile these features into ES2017 with await/async, perhaps even with the new [Asynchronous Iteration](https://github.com/tc39/proposal-async-iteration) proposal which is already [stage 3](https://github.com/tc39/proposals).

**Comment by alangpierce** [comment]
Makes sense, I've certainly heard of ICS, although I've never used it. After looking at it and thinking a little, looks like there would be some challenges:
* ICS is a fork of CoffeeScript based on an older version, which probably would complicate parsing. Currently, decaffeinate uses the CoffeeScript parser as the first step, then reads the CoffeeScript AST and transforms it into the decaffeinate AST. Even if there was an option to use the ICS parser, I worry that the AST format may have changed (the CoffeeScript project certainly doesn't guarantee a stable AST format). So getting this to work may require merging any CS parser updates into ICS, or porting just the parser changes from ICS into the [decaffeinate-coffeescript](https://github.com/decaffeinate/coffeescript) project. Also, regardless, there would need to be a flag for whether the code is ICS, since ICS is technically not a strict superset of CS. For example `await = 1` appears to be valid CS but invalid ICS.
* From what I can tell, the `await`/`defer` syntax in ICS doesn't actually map in any straightforward way to JS `async`/`await`. Certainly they have conceptual similarities, but, for example, it looks like ICS assumes that your code uses callback and `async`/`await` uses promises for everything.

Also, do you know any big open source projects using ICS?

**Comment by niieani** [comment]
@alangpierce Indeed, the second point you mention is a bit of a challenge - there isn't an exact mapping and it would take some effort to figure out the most similar/meaningful/equivalent output based on the ICS input.

As for big open source projects, the biggest users of ICS as far as I know are @keybase. Some huge OSS projects of theirs include:
- [kbpgp](https://github.com/keybase/kbpgp) - PGP implementation in ICS
- [triplesec](https://github.com/keybase/triplesec) - triple-symmetric key encryption of smallish secrets

Both pretty good and useful projects, _kbpgp_ being much faster than _OpenPGP.js_. I don't think they're getting much love from the world cause of their use of ICS, I think the projects would get more contributions had they been deiced-decaffeinated.
